src/App.tsx

```diff
@@ -1,26 +1,390 @@
 /* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useEffect, useRef } from 'react';
+import { getTodos, createTodo, deleteTodo, updateTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
 import { UserWarning } from './UserWarning';
+import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { Footer } from './components/Footer';
+import FilterStatus from './enums/FilterStatus';
 
-const USER_ID = 0;
+const USER_ID = 2338;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.All,
+  );
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  const [updatingTodoId, setUpdatingTodoId] = useState<number | null>(null);
```

Comment:
Users cannot delete and update todo at the moment. Only one action is possible. So you can get rid of one state and rename another to `loadingTodoId`